libafl_bolts/src/shmem.rs

```diff
@@ -204,52 +204,24 @@ pub trait ShMem: Sized + Debug + Clone + AsSlice<Entry = u8> + AsMutSlice<Entry
         self.len() == 0
     }
 
-    /// Convert to an owned object reference
-    ///
-    /// # Safety
-    /// This function is not safe as the object may be not initialized.
-    /// The user is responsible to initialize the object with something like
-    /// `*shmem.as_object_mut::<T>() = T::new();`
-    unsafe fn as_object<T: Sized + 'static>(&self) -> &T {
-        assert!(self.len() >= core::mem::size_of::<T>());
-        (self.as_slice().as_ptr() as *const () as *const T)
-            .as_ref()
-            .unwrap()
-    }
-
-    /// Convert to an owned object mutable reference
-    ///
-    /// # Safety
-    /// This function is not safe as the object may be not initialized.
-    /// The user is responsible to initialize the object with something like
-    /// `*shmem.as_object_mut::<T>() = T::new();`
-    unsafe fn as_object_mut<T: Sized + 'static>(&mut self) -> &mut T {
-        assert!(self.len() >= core::mem::size_of::<T>());
-        (self.as_mut_slice().as_mut_ptr() as *mut () as *mut T)
-            .as_mut()
-            .unwrap()
-    }
-
-    /// Convert to a slice of type &\[T\]
-    ///
-    /// # Safety
-    /// This function is not safe as the object may be not initialized.
-    /// The user is responsible to initialize the objects in the slice
-    unsafe fn as_objects_slice<T: Sized + 'static>(&self, len: usize) -> &[T] {
-        assert!(self.len() >= core::mem::size_of::<T>() * len);
-        let ptr = self.as_slice().as_ptr() as *const () as *const T;
-        core::slice::from_raw_parts(ptr, len)
-    }
-
-    /// Convert to a slice of type &mut \[T\]
-    ///
-    /// # Safety
-    /// This function is not safe as the object may be not initialized.
-    /// The user is responsible to initialize the objects in the slice
-    unsafe fn as_objects_slice_mut<T: Sized + 'static>(&mut self, len: usize) -> &mut [T] {
-        assert!(self.len() >= core::mem::size_of::<T>() * len);
-        let ptr = self.as_mut_slice().as_mut_ptr() as *mut () as *mut T;
-        core::slice::from_raw_parts_mut(ptr, len)
+    /// Convert to a ptr of a given type, checking the size.
+    /// If the map is too small, returns `None`
+    unsafe fn as_ptr<'a, T: Sized + 'static>(&self) -> Option<*const T> {
+        if self.len() >= core::mem::size_of::<T>() {
+            Some(self.as_slice().as_ptr() as *const T)
+        } else {
+            None
+        }
+    }
+
+    /// Convert to a mut ptr of a given type, checking the size.
+    /// If the map is too small, returns `None`
+    fn as_mut_ptr_of<T: Sized>(&mut self) -> Option<*mut T> {
```

Comment:
This one has the suffix `_of`, but `as_ptr` doesn't - maybe remove it here or add it there?

Reply:
think maybe both should have it, since it's casting to a specific pointer (?)

Reply:
renamed to _of, thx!